### PR TITLE
redirect should add mountPath prefix

### PR DESCRIPTION
**Pull request by coderhaoxin**
Maybe it's not a good idea to do this in koa, any thought?

### CODE REVIEW

File: lib/response.js [hunk 1]
```diff
@@ -251,7 +251,9 @@ module.exports = {
 
   redirect: function(url, alt){
     // location
+    const mountPath = this.ctx.mountPath;
```
Comment (dead-horse) [inline lib/response.js]: -1.

override `redirect` in `koa-mount` is much better than add this logic here, or write a `koa-mount-redirect` for this feature alone.

**Comment by juliangruber** [comment]
-1 from me as well, at least for this implementation. Does anyone know how express handles this?

I'm thinking either koa-mount should patch .redirect, or koa should offer a more generic way to handle this without patching.

**Comment by juliangruber** [comment]
actually I'm also -1 to this logic, if I do `.redirect('/foo')` i expect it to redirect to `DOMAIN/foo`, instead of `DOMAIN/prefix/foo`. That's what absolute paths should do. Otherwise, use relative paths.

**Comment by juliangruber** [comment]
I'd suggest adding a method like `.subRedirect(path)` via your middleware that contains this logic.

**Comment by dougwilson** [comment]
> Does anyone know how express handles this?

Express does pretty much the same as Koa, both just passing through whatever URL was provided.

**Comment by coderhaoxin** [comment]
> override redirect in koa-mount is much better than add this logic here, or write a koa-mount-redirect for this feature alone.

It's not a good idea to override method in middlewares IMO, also not worth to write a new module.

> I'd suggest adding a method like .subRedirect(path) via your middleware that contains this logic.

+1 for this!

**Comment by coderhaoxin** [comment]
> I'd suggest adding a method like .subRedirect(path) via your middleware that contains this logic.

Actually, I think relative path is enough. https://github.com/koajs/mount/issues/45

**Comment by peey** [comment]
I don't think this should be done in koa. 

Overriding in koa-mount would be a better solution and more in line with the behavior on how it already overrides the original path

**Pull request closed without merging**